clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -65,7 +65,7 @@
     public static final long INVALID_HIGHWATERMARK = -1L;
     public static final ByteBuffer EMPTY_RECORD_SET = ByteBuffer.allocate(0);
 
-    private final Map<TopicPartition, PartitionData> responseData;
+    private final LinkedHashMap<TopicPartition, PartitionData> responseData;
```

Comment:
Is there a specific reason to preserve the order? I'm guessing we'll end up looking up by key anyway.